ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TxStats.java

```diff
@@ -184,9 +204,37 @@ void incrementAbortedTransactions()
         numberOfAbortedTransactions.incrementAndGet();
 	}
 
+	/**
+	 * @Deprecated as of 5.0.5.Final use {@link #incrementCommittedTransactions(long)}} instead
+	 */
+	@Deprecated
 	void incrementCommittedTransactions()
 	{
-        numberOfCommittedTransactions.incrementAndGet();
+        incrementCommittedTransactions(0L);
+	}
+
+	/**
+	 * @param duration the time in nanoseconds it took for the 2PC phase to complete. The averaged commit
+	 *                   time is available by calling {@link #getAverageCommitTime()}
+	 */
+	void incrementCommittedTransactions(long duration) {
+		TxCommitStatistic prev, next;
+
+		if (duration == 0)
+			duration = 1; // a duration of zero is disallowed - round up to minimum unit
+
+		do {
+			prev = commitStatistic.get();
+
+			long prevCount = prev.getNumberOfCommittedTransactions();
```

Comment:
looks good in general - can't we just call getNumberOfCommittedTransactions instead of storing it in TxCommitStatistic? Then you could just an atomicreference to a long rather than the new TxCommitStatistic. Overall my question is to the need for TxCommitStatistic instead of just averageCommitTime stored as a long